lib/js/components/Switch/Switch.vue

```diff
@@ -0,0 +1,350 @@
+			border-color: $color-primary-border-disabled;
+		}
+	}
+
+	&.-disabled &__icon {
+		color: $color-neutral-icon-disabled;
+	}
+
+	&.-disabled .-selected &__icon {
+		color: $color-primary-icon-disabled;
+	}
+
+	&.-disabled &__label {
+		color: $color-neutral-text-weak;
+	}
+
+	&.-disabled .-selected &__label {
+		color: $color-neutral-text-strong;
+	}
+
+	&:not(.-disabled) &__item:not(.-selected) {
+		&:hover {
+			background-color: $color-neutral-background-ghost-hovered;
+
+			&::before {
+				background-color: $color-neutral-ripple;
+				border-radius: 50%;
+				content: '';
+				padding-top: 100%;
+				position: absolute;
+				transform: scale(0);
+				transition: none;
+				width: 100%;
+			}
+		}
+
+		&:active {
+			&::before {
+				content: '';
+				position: absolute;
+				transform: scale(1);
+				transition: $switch-transition;
+			}
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import { PropType, toRaw } from 'vue';
+import {
+	SWITCH_RADIUSES,
+	SWITCH_SIDE,
+	SWITCH_SIZES,
+	SWITCH_STATE,
+	SwitchRadius,
+	SwitchSelection,
+	SwitchSize,
+	SwitchState,
+} from './Switch.consts';
+import DsIcon, { ICON_SIZES, ICONS, IconItem } from '../Icons/Icon';
+
+export default {
+	// eslint-disable-next-line vue/no-reserved-component-names
+	name: 'Switch',
+	components: {
+		DsIcon,
+	},
+	props: {
+		size: {
+			type: String as PropType<SwitchSize>,
+			default: SWITCH_SIZES.MEDIUM,
+			validator(size: SwitchSize) {
+				return Object.values(SWITCH_SIZES).includes(size);
+			},
+		},
+		radius: {
+			type: String as PropType<SwitchRadius>,
+			default: SWITCH_RADIUSES.CAPSULE,
+			validator(value: SwitchRadius) {
+				return Object.values(SWITCH_RADIUSES).includes(value);
+			},
+		},
+		iconLeft: {
+			type: Object as PropType<IconItem>,
+			default: null,
+			validator(icon: IconItem) {
+				return Object.values(ICONS).includes(toRaw(icon));
+			},
+		},
+		iconRight: {
+			type: Object as PropType<IconItem>,
+			default: null,
+			validator(icon: IconItem) {
+				return Object.values(ICONS).includes(toRaw(icon));
+			},
+		},
+		labelLeft: {
+			type: String,
+			default: null,
+		},
+		labelRight: {
+			type: String,
+			default: null,
+		},
+		state: {
+			type: String as PropType<SwitchState>,
+			default: SWITCH_STATE.DEFAULT,
+			validator(state: SwitchState) {
+				return Object.values(SWITCH_STATE).includes(state);
+			},
+		},
+		selectedSide: {
+			type: String as PropType<SwitchSelection>,
+			default: SWITCH_SIDE.LEFT,
+			validator(side: SwitchSelection) {
+				return Object.values(SWITCH_SIDE).includes(side);
+			},
+		},
+	},
+	emits: ['update:selectedSide'],
+	data() {
+		return {
+			ICONS: Object.freeze(ICONS),
+			ICON_SIZES: Object.freeze(ICON_SIZES),
+			SWITCH_RADIUSES: Object.freeze(SWITCH_RADIUSES),
+			SWITCH_SIDE: Object.freeze(SWITCH_SIDE),
+			SWITCH_SIZE: Object.freeze(SWITCH_SIZES),
+			SWITCH_STATE: Object.freeze(SWITCH_STATE),
+			currentSide: this.selectedSide,
```

Comment:
Let's not rewrite the side but always use side that comes from props.

Reply:
I fully agree with that. However, there is a problem with storybook (known issue). Once you change value through the props (controls) the storybook iframe is being refreshed and it breaks the animation preview. Also, I don't like storybook-smelling code in the production component, so I can address it after testing. What do you think?

Alternatively, it looks like is has been addressed in the newer version of storybook. But upgrading DB doesn't seem to be in scope of this ticket.

Reply:
We decided on daily that this is fine until we upgrade storybook to a newer version. Let's leave a comment here